hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c

```diff
@@ -0,0 +1,500 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <hal/hal_bsp.h>
+#include <assert.h>
+#include <os/os.h>
+#include <errno.h>
+#include <pwm/pwm.h>
+#include <string.h>
+
+/* Nordic headers */
+#include <bsp.h>
+#include <nrf.h>
+#include <nrf_pwm.h>
+#include <nrf_drv_pwm.h>
+#include <nrf_drv_clock.h>
+/* #include <bsp/cmsis_nvic.h> */
+#include <app_timer.h>
+
+//#include <app_error.h>
```

Comment:
Minor nit. We'd prefer /* */ style comments over //. I believe this is listed in coding style doc.